components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -55,10 +55,23 @@ namespace ai_chat {
 
 namespace {
 
-constexpr char16_t kYoutubeTranscriptUrlExtractionScript[] =
+constexpr char16_t kYoutubeInnerTubeConfigExtractionScript[] =
     uR"JS(
       (function() {
-        return ytplayer?.config?.args?.raw_player_response?.captions?.playerCaptionsTracklistRenderer?.captionTracks
+        // Get API key from ytcfg or fallback to regex
+        const apiKey = ytcfg?.data_?.INNERTUBE_API_KEY ||
+                      document.querySelector('script')
```

Comment:
Does this always necessarily match the correct script element? We don't need to iterate through all of them?
What are the circumstances where we don't have the data on the ytcfg object? Is it video player page vs list page?

Reply:
`ytcfg.data_.INNSERTUBE_API_KEY` is always available and the same on my client (even in private window) regardless it is a video page or video list or just youtube home page. Same as that script element
<img width="954" alt="Screenshot 2025-06-17 at 20 11 03" src="https://github.com/user-attachments/assets/5bea4d0e-d836-49cd-a216-2e9a5ae5e0e5" />
Maybe we don't need this regex fallback because the script is basically set the object.

Reply:
My main concern is using `document.querySelector` for the fallback and not `document.querySelectorAll` as 1) `querySelector` will not find what it's looking for if it's not in the first `<script>` element and 2) this code will throw an error if there are no matching `<script>` tags at all (less likely)

Reply:
fixed in https://github.com/brave/brave-core/pull/29619/commits/751bd2bdb0242311e077cd87d5df41c3b7e530cb